src/Emulator/Visuals/Visuals.tsx

```diff
@@ -48,13 +48,26 @@ const HighlightButton: FC<BtnProps> = ({ children, hoverComponent, component, ..
 const Visuals = () => {
   const { sessionId } = useEmulator();
   const { data } = useVisualizeToynetImage(sessionId);
+
   const [pos, setPos] = useState({x: 0, y: 0});
```

Comment:
Then this can become

```jsx
useEffect(() => {
    if (!isLoading) {
        recenterImage();
    }
}, [isLoading])
```

it's probably not the best idea to use a ref in a useEffect since they don't trigger re-renders